llvm/lib/MC/DXContainerRootSignature.cpp

```diff
@@ -46,6 +48,16 @@ size_t RootSignatureDesc::getSize() const {
         Size += sizeof(dxbc::RTS0::v2::RootDescriptor);
 
       break;
+    case llvm::to_underlying(dxbc::RootParameterType::DescriptorTable):
+      const DescriptorTable &Table =
+          ParametersContainer.getDescriptorTable(I.Location);
+      if (Version == 1)
+        Size +=
+            sizeof(dxbc::RTS0::v1::DescriptorRange) * Table.Ranges.size() + 8;
+      else
+        Size +=
+            sizeof(dxbc::RTS0::v2::DescriptorRange) * Table.Ranges.size() + 8;
```

Comment:
```suggestion
      if (Version == 1)
        Size +=
            sizeof(dxbc::RTS0::v1::DescriptorRange) * Table.Ranges.size();
      else
        Size +=
            sizeof(dxbc::RTS0::v2::DescriptorRange) * Table.Ranges.size();
      Size += sizeof(uint32_t); // Ranges offset
```
I think the extra space is for the offset correct?

Reply:
4 bits for offset, and 4 bits for the amount of ranges in the table.